### PR TITLE
PIM-5210: Respect current mass edit common attributes behavior

### PR DESCRIPTION
**Pull request by grena**
| Q                 | A
| ----------------- | ---
| Specs             | N
| Behats            | N
| Blue CI           |
| Changelog updated | N
| Review and 2 GTM  | Y

### CODE REVIEW

File: features/Context/Page/Base/ProductEditForm.php [hunk 1]
```diff
@@ -559,4 +560,184 @@ public function findValidationTooltip($text)
             );
         });
     }
+
+    /**
+     * Check if the specified field is set to the expected value, raise an exception if not
```
Comment (rybus) [inline features/Context/Page/Base/ProductEditForm.php]: Check**s** [...] raise**s** :D

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php [hunk 2]
```diff
@@ -354,4 +359,20 @@ public function getErrors()
     {
         return $this->errors;
     }
+
+    /**
+     * @return string
+     */
+    public function getCurrentLocale()
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php]: why public if you only use it here?

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php [hunk 3]
```diff
@@ -354,4 +359,20 @@ public function getErrors()
...
+    {
+        return $this->currentLocale;
+    }
+
+    /**
+     * @param string $currentLocale
+     */
+    public function setCurrentLocale($currentLocale)
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php]: why public if you only use it here?

File: features/Context/Page/Base/ProductEditForm.php [hunk 4]
```diff
@@ -559,4 +560,184 @@ public function findValidationTooltip($text)
...
+                    $expected,
+                    $actual
+                ),
+                $this->getSession()
+            );
+        }
+    }
+
+    /**
+     * Return the current value of a textarea
```
Comment (rybus) [inline features/Context/Page/Base/ProductEditForm.php]: Return**s**

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php (hunk 3 above)
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/EditCommonAttributes.php]: It's used by the form to set locale on form submit

File: features/mass-action/edit_common_attributes_locale_specific.feature [hunk 5]
```diff
@@ -25,3 +25,31 @@ Feature: Edit common attributes of many products at once with locale specific ca
...
+    Then the product Description should be empty
+    And I visit the "Additional information" group
+    But the product Weight should be "10 KILOGRAM"
+    And I move on to the next step
+    And I wait for the "edit-common-attributes" mass-edit job to finish
+    And the product "tshirt" should have the following values:
+      | description-en_US-tablet    |                  |
+      | description-en_US-ecommerce |                  |
+      | description-fr_FR-ecommerce | Mon joli tshirt  |
+      | weight                      | 10.0000 KILOGRAM |
```
Comment (pierallard) [inline features/mass-action/edit_common_attributes_locale_specific.feature]: Given / When / Then

**Comment by skeleton** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/mass-edit/hidden-field-updater.js [hunk 6]
```diff
@@ -30,11 +35,31 @@ define(
...
+                data = _.mapObject(data, function (attributeValues) {
+                    var values = _.map(attributeValues, function (value) {
+                        if (null !== value.locale && selectedLocale !== value.locale) {
+                            value.data = null;
+                        }
+
+                        return value;
+                    });
+
+                    return values;
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/mass-edit/hidden-field-updater.js]: why not a direct `return _.map(...)` ?

**Comment by pierallard** [comment]
Tiny questions then GTM

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/mass-edit/hidden-field-updater.js (hunk 6 above)
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/mass-edit/hidden-field-updater.js]: :+1:

**Pull request merged**